onnxruntime/core/providers/openvino/backends/basic_backend.cc

```diff
@@ -26,56 +27,142 @@ BasicBackend::BasicBackend(const ONNX_NAMESPACE::ModelProto& model_proto,
+        }
+        else {
+            model_blob_name = global_context_.onnx_model_name + "_" + "op_v_" + std::to_string(global_context_.onnx_opset_version) + "_" + model_name.substr(0,model_extension_index) + "_" + hw_target + "_" + subgraph_context_.subgraph_name + "_ov_" + "partially" + ".blob";
+        }
+        if(global_context_.blob_dump_path == "" || global_context_.blob_dump_path == "\"" || global_context_.blob_dump_path.empty()) {
+          ov_compiled_blobs_dir = openvino_ep::backend_utils::GetCurrentWorkingDir() + "/ov_compiled_blobs/";
+        } else {
+          ov_compiled_blobs_dir = global_context_.blob_dump_path + "/ov_compiled_blobs";
+        }
+        if(openvino_ep::backend_utils::IsDirExists(ov_compiled_blobs_dir)) {
+          LOGS_DEFAULT(INFO) << log_tag << "'ov_compiled_blobs' directory already exists at the executable path";
+        }
+        else {
+          CreateDirectory(ov_compiled_blobs_dir);
+        }
+        blob_path.open(ov_compiled_blobs_dir + "/" + model_blob_name);
+        if (!blob_path.is_open()) {
+            LOGS_DEFAULT(INFO) << log_tag << "Device specific Compiled blob doesn't exist for this model";
+        } else {
+            LOGS_DEFAULT(INFO) << log_tag << "Device specific Compiled blob already exists for this model";
+            vpu_status = true;
+        }
+    }
+  }
 
-#if defined(OPENVINO_2020_4) || defined(OPENVINO_2021_1) || defined(OPENVINO_2021_2)
-  if (const_outputs_map_.size() == subgraph_context_.output_names.size())
-    subgraph_context_.is_constant = true;
-#endif
+  //validate const subgraphs
+  if(!openvino_ep::BackendManager::GetGlobalContext().is_wholly_supported_graph) {
+    ie_cnn_network_ = CreateCNNNetwork(model_proto, global_context_, subgraph_context_, const_outputs_map_);
+    SetIODefs(model_proto, ie_cnn_network_, subgraph_context_.output_names, const_outputs_map_, global_context_.device_type);
+  #if defined(OPENVINO_2020_4) || defined(OPENVINO_2021_1) || defined(OPENVINO_2021_2)
+    if (const_outputs_map_.size() == subgraph_context_.output_names.size())
+      subgraph_context_.is_constant = true;
+  #endif
 
   // Loading model to the plugin
-  if (subgraph_context_.is_constant)
-    return;
-  std::map<std::string, std::string> config;
-#ifndef NDEBUG
-  if (openvino_ep::backend_utils::IsDebugEnabled()) {
-    config["PERF_COUNT"] = CONFIG_VALUE(YES);
+    if (subgraph_context_.is_constant) {
+      LOGS_DEFAULT(INFO) << log_tag << "The subgraph is a const. Directly moving to Infer stage.";
+      return;
+    }
   }
-#endif
-  if (global_context_.device_type.find("MYRIAD") != std::string::npos) {
-#if defined(OPENVINO_2021_1) || defined(OPENVINO_2021_2)
-    if (subgraph_context_.set_vpu_config) {
-      config["MYRIAD_DETECT_NETWORK_BATCH"] = CONFIG_VALUE(NO);
+
+  char* compiled_blob_path = nullptr;
+  if (vpu_status == true || openvino_ep::backend_utils::UseCompiledNetwork()) {
+  #ifdef _WIN32
+		size_t env_name_len = 0;
```

Comment:
indents look off here.
make sure code conforms to .clang-format in root of repo.